src/main/java/net/earthcomputer/clientcommands/command/MapCommand.java

```diff
@@ -0,0 +1,93 @@
+import net.minecraft.client.texture.NativeImage;
+import net.minecraft.client.util.ScreenshotRecorder;
+import net.minecraft.item.FilledMapItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.map.MapState;
+import net.minecraft.text.ClickEvent;
+import net.minecraft.text.Text;
+
+import java.io.File;
+import java.io.IOException;
+
+import static com.mojang.brigadier.arguments.IntegerArgumentType.getInteger;
+import static com.mojang.brigadier.arguments.IntegerArgumentType.integer;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.argument;
+
+public class MapCommand {
+    private static final SimpleCommandExceptionType NO_HELD_MAP = new SimpleCommandExceptionType(Text.translatable("commands.cmap.noHeld"));
+
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("cmap")
+            .then(literal("export")
+                .executes(ctx -> exportMap(ctx.getSource(), 1))
+                .then(argument("upscale", integer(1, 16))
+                    .executes(ctx -> exportMap(ctx.getSource(), getInteger(ctx, "upscale")))
+                )
+            )
+        );
+    }
+
+    private static MapState getMapState(ClientPlayerEntity player) throws CommandSyntaxException {
+        ItemStack map;
+
+        // detect if map in hand
+        if (player.getMainHandStack().getItem() instanceof FilledMapItem) {
+            map = player.getMainHandStack();
+        } else if (player.getOffHandStack().getItem() instanceof FilledMapItem) {
+            map = player.getOffHandStack();
+        } else {
+            throw NO_HELD_MAP.create();
+        }
+
+        Integer mapId = FilledMapItem.getMapId(map);
+        MapState mapState = FilledMapItem.getMapState(mapId, player.world);
+        if (mapState == null) {
+            throw NO_HELD_MAP.create();
+        }
+        return mapState;
+    }
+
+
+    private static int exportMap(FabricClientCommandSource source, int upscale) throws CommandSyntaxException {
+        MapState mapState = getMapState(source.getPlayer());
+        try (NativeImage image = new NativeImage(NativeImage.Format.RGBA, 128 * upscale, 128 * upscale, false)) {
+            for (int i = 0; i < 128 * upscale; i+= upscale) {
+                for (int j = 0; j < 128 * upscale; j+= upscale) {
+                    int x = i / upscale;
+                    int y = j / upscale;
+                    int color = MapColor.getRenderColor(mapState.colors[x + y * 128]);
+                    for (int k = 0; k < upscale; k++) {
+                        for (int l = 0; l < upscale; l++) {
+                            image.setColor(i + k, j + l, color);
+                        }
+                    }
+                }
+            }
+            File screenshotDir = new File(MinecraftClient.getInstance().runDirectory, "screenshots");
```

Comment:
You can use `source.getClient()` here instead.

Reply:
Actually, what about saving this in a new directory in `config/clientcommands` instead? For that you can use `ClientCommands.configPath`.

Reply:
Nor are `.mcfunction` files, or any of the other files we store there really. The `config` directory isn't bound to just configuration files imo.

Reply:
I would consider the functions to be a form of configuration, since they are generally persisted between launches and control functionality in some way. These exported images are not directly used by the game.

Reply:
^